apps/antalmanac/src/stores/AppStore.ts

```diff
@@ -6,11 +6,17 @@ import { Schedules } from './Schedules';
 import { SnackbarPosition } from '$components/AppBar/NotificationSnackbar';
 import { CalendarEvent, CourseEvent } from '$components/Calendar/CourseCalendarEvent';
 import { RepeatingCustomEvent } from '$components/Calendar/Toolbar/CustomEventDialog/CustomEventDialog';
+import { useTabStore } from '$stores/TabStore';
+
+function getCurrentTheme() {
+    const theme = typeof Storage === 'undefined' ? 'auto' : window.localStorage.getItem('theme');
+    return theme === null ? 'auto' : theme;
+}
 
 class AppStore extends EventEmitter {
     schedule: Schedules;
     customEvents: RepeatingCustomEvent[];
-    colorPickers: { [key: string]: EventEmitter };
+    colorPickers: Record<string, EventEmitter>;
```

Comment:
In general, I like to use `Record` if possible because it's more succinct. A niche edge case where this isn't very nice is when you want to map an interface's properties to alternate values. A `Record` isn't able to capture the original definition of the properties.

[Here's an example on TS playground](https://www.typescriptlang.org/play?ssl=3&ssc=12&pln=2&pc=12#code/C4TwDgpgBAsiCSA7YEBOAzAhgY2gXigG8AoKKTALigGdhUBLRAc1KgCMraHnXtO7GLAL7FioSLBAAlCNgD2qACZQCM+UoA8Aawgg56SUhQYcEADQ0BzAHyiANhGBQAHlThqFyggFdEiiOiMEMqY1OSIIKIA9FFQAKrUglAARExyUMDp-oGI9MD0cojJUIUZABbQyZjFYKhykKigUIpyENSIAOROTI5Qet7hIADuFagQxDHsspje1NDAFVAABh5KSxng0ABWs07oCj3AYSNo84sK9EyMmHZQtfVo+W1Q2JgAttDodW8Tsb7+qHK0DKcjkigseQ6YTY01m0DyUB2tBehS43mwR3IUF8BUQJQMN1uC2gOhA1AhiFoEEwyn05Ax3kJICgb0wYDASS+cjeQJKDCuiBuUFJxGcADpMKJxNA4DA2ZBFEY0FhcCoiFAANoAaSgjGFujpcCVJlwAF1+NwmFARMQHE4QG4QHL2cFjSr8Ni-AEgiEwpgItFYgkkql0plmt7cvlCsVSsSUtU7nUGk1gJgdGF+uwcFoMukjchlaYNR1MB1Tb8pq84VjWS7lGN1A2IB83jDUGF4xcBUL7inmQihqEIwwAG7BKBcn4gCVAA)

In GitHub markdown:
```ts
type MyInterface = {
  a: string
  b: string
  c: string
}

type MyRecord = Record<keyof MyInterface, string>

let x: MyRecord = undefined as any

// Using "go to definition" on the "a" property doesn't get you anywhere
// because the `Record` type just forgets where the original properties came from
// under the hood, it's because it just constructs a union of all the keys, instead of actually mapping from the original key
x.a

type MyMappedInterface = { [K in keyof MyInterface]: string }

let y: MyMappedInterface = undefined as any

// Using "go to definition" on the "a" property takes you back to MyInterface['a']
// because a mapped record remembers the original property it was derived from
y.a
```

Reply:
I don't disagree, but I'd argue that it's out of scope. I'll leave it since you've already done it.